### PR TITLE
examples: Add step-by-step NMO correction using SparseFunctions.

### PR DESCRIPTION
**Pull request by speglich**
This example is a Normal Moveout (NMO) correction, that shows how Devito is more than a Finite Difference solver, and could help some users a better understanding of Devito.

### CODE REVIEW

File: examples/seismic/tutorials/10_nmo_correction.ipynb [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
...GrY3rHauFUHnJYzpA2z3DLz/88etrbd9K2dnX11dWVlak/6oai8WKi4u/9vc1PL4qhRIvC4WABP1QeQJta6soL1d7/tUTh9cqXaPu0u8xqDIIj7SsUnvduRBiw5MvsQBSKEGp0OuJfsjowXzrx7s9o+EtURktsS2jpsZT51wNXx1Z9Ebt7W3zVK/m28Ky7hwSolo/28qLD0WxVtTYvnxk+0fr70rf77DHWAaJRCKhUOjeecvQDwdWvOVlAhL0QwgxOBJfHDkRae9TcBGIccuQ5maFvWBcobf8599TFLq7hChWuwYy0rLqo/V3jZ3/QGEvbFz7ysoljSe7exTWgaKDQECafpx46paQ/V/VTt+Dzrd+CCFU3rFR9cK99fPzRn/nrdvmNJ3cat3pye0G/WwrZXe6nCqTi/s2nHtm7rTPDejo6FhX/+NDbxydarmkh4AXCMjRD6OlOw59NP/p7oaW0/u7h6W2PRrV1kCKi7U78A6pWQe4cu3Kuvc23rptzq3b5qx7b6P0C/b69AvwioVoFELN05YuH9neF775fGT+5SPbpfa+pbCT3T0rlzTWVjXsat4/fOFjyztsQiBnBKTph6XGO6tCoVueOmHZ4/dNCfphGMgT23oUXIBeWira25X3oXFf3dvb5kmfsaba9Hb9N7aagjjbSghxIx4bfvnhoRcWTbWdKemNJ+kuurtuzcpNu5r3f/h+b0oCXkLAswRk6ocBYX/3cEPL6flPd6/f3Sv1durd3aK6WrsXVkWFULcYcnz4/eU//95nfnzr7W3zNh5rkhvVu/V16VlClKu6Hdalg1s+Wn9X77JPD72wKN6v5unjvT2Dm9e1LLq7rn75xn1th3gQoWejU4FWDP2Q0XFy9EMI0Xli5L4NxxtaTndFL8pomFFGY6M2XSn9G746Mr994Zp31imtxaSF9+nzWakHH+iRqepj5z8YXFPm8nqPTJlY9w9f+PjQgaOb17WsWLh2wdxH65dvbN60m5OyrIjY9iAB+fphQBi9cm3X4fPhLdEFP/zF+t29vzh7WR6cpiZtMWTGDO3/TcpuSvtGf+dDb4Zv3jL7q+0L5bV9vKRmISqEmKH/v0kI2Q8CvzbSf3Hfhv7HSvrCNw+//HD89BHpBLQC3z54bMOTL917x8o1Kze17+TWWEo6wYeFekI/9BOzLOdgpSyPpLytvWv+WY5K652
```
Comment (EdCaunt) [inline examples/seismic/tutorials/10_nmo_correction.ipynb]: This is a default iirc. Is it worth dropping for cleanliness? I suppose it lets the user know that they can set a non-zero origin.

Comment (EdCaunt) [inline examples/seismic/tutorials/10_nmo_correction.ipynb]: "encapsulates these properties"

Comment (mloubout) [inline examples/seismic/tutorials/10_nmo_correction.ipynb]: common midpoint gather (CMP)

Comment (EdCaunt) [inline examples/seismic/tutorials/10_nmo_correction.ipynb]: Maybe want to amend the wording of the grid points? I'm assuming you mean 10 total, but the ambiguity might be slightly confusing for users. I suppose look at previous notebooks for the precedent.

Comment (mloubout) [inline examples/seismic/tutorials/10_nmo_correction.ipynb]: I'd do 
shape = (301, 501)
spacing = (10., 10.)

THis makes nicer data, and it's still small enough to run fast.

Comment (mloubout) [inline examples/seismic/tutorials/10_nmo_correction.ipynb]: Is that long enough?

Comment (mloubout) [inline examples/seismic/tutorials/10_nmo_correction.ipynb]: nbl=40 for nice data
space_order can be removed here for acoustic, if not use the same as you used for `u`

Comment (EdCaunt) [inline examples/seismic/tutorials/10_nmo_correction.ipynb]: Alternative wording: "The receivers start at (2000, 20) with an even horizontal spacing of 100m at consistent depth."

Comment (mloubout) [inline examples/seismic/tutorials/10_nmo_correction.ipynb]: sace_order=4 for nicer data

Comment (mloubout) [inline examples/seismic/tutorials/10_nmo_correction.ipynb]: Receivers are basically free to model so you can do `nrcv=251` or something to have the classical hyperbolic even show up nicely in the plot.

Comment (mloubout) [inline examples/seismic/tutorials/10_nmo_correction.ipynb]: wouldn't it be better with the first receiver at the source location:
`np.linspace(src.coordinates.data[0, 0], model.domain_size[0], num=nrcv)`

Comment (EdCaunt) [inline examples/seismic/tutorials/10_nmo_correction.ipynb]: Maybe change the t_0/t_end variable names to match the other tutorials?

Comment (mloubout) [inline examples/seismic/tutorials/10_nmo_correction.ipynb]: add a scaling to have better visualization:
```
scale = np.max(rec)/100
plot = plt.imshow(rec, vmin=-scale, vmax=scale, ...)
```

Comment (EdCaunt) [inline examples/seismic/tutorials/10_nmo_correction.ipynb]: You can also add the NBVAL_IGNORE_OUTPUTs as cell tags to tidy up the cells

Comment (EdCaunt) [inline examples/seismic/tutorials/10_nmo_correction.ipynb]: "Auxiliary function for plotting traces:"

Comment (EdCaunt) [inline examples/seismic/tutorials/10_nmo_correction.ipynb]: "If our model wasn't purely horizontal, we would have to sort these traces by common midpoints prior to NMO correction."

Comment (EdCaunt) [inline examples/seismic/tutorials/10_nmo_correction.ipynb]: t and x want to be formatted in the same way as t_0 here I think.

Comment (EdCaunt) [inline examples/seismic/tutorials/10_nmo_correction.ipynb]: "This equation results from the Pythagorean theorem, and is only valid for horizontal reflectors."

Comment (EdCaunt) [inline examples/seismic/tutorials/10_nmo_correction.ipynb]: Alternative wording "grid of size samples x traces."

Comment (EdCaunt) [inline examples/seismic/tutorials/10_nmo_correction.ipynb]: Probably drop "this is a very simple approach".

Comment (EdCaunt) [inline examples/seismic/tutorials/10_nmo_correction.ipynb]: "as the grid"

Comment (EdCaunt) [inline examples/seismic/tutorials/10_nmo_correction.ipynb]: "as the grid, describing the"

Comment (EdCaunt) [inline examples/seismic/tutorials/10_nmo_correction.ipynb]: Possibly change the naming convention for these equations?

**Comment by mloubout** [comment]
And the `NBVAL_IGNORE_OUTPUTs`  in the `op()` cell to avoid testing erroring from difference in runtime

**Pull request merged**